app/Http/Controllers/Api/V1/ActivityController.php

```diff
@@ -627,14 +632,25 @@ public function getH5pResourceSettingsShared(Activity $activity)
     private function hasPermission(Activity $activity)
     {
         $authenticated_user = auth()->user();
-        $project_users = $activity->playlist->project->users;
-
+        $project = $activity->playlist->project;
+        $project_users = $project->users;
         foreach ($project_users as $project_user) {
             if ($authenticated_user->id === $project_user->id) {
                 return true;
             }
         }
 
+        $project_teams = $project->teams;
+        foreach ($project_teams as $project_team) {
+            $team_project_user = TeamProjectUser::where('team_id', $project_team->id)
+                ->where('project_id', $project->id)
+                ->where('user_id', $authenticated_user->id)
+                ->first();
```

Comment:
Just a suggestion here as you are using fetched data for only existence. You might replace first() with exists()